File: storio-sqlite-processor/src/main/java/com/pushtorefresh/storio/sqlite/processor/StorIOSQLiteTypeMeta.java
package com.pushtorefresh.storio.sqlite.processor;

import java.util.HashMap;
import java.util.Map;

public class StorIOSQLiteTypeMeta {

    public final String simpleName;
    public final String packageName;
    public final String tableName;

    /**
     * Yep, this is MODIFIABLE Map, please use it carefully
     */
    public final Map<String, StorIOSQLiteTypeMeta> columns = new HashMap<String, StorIOSQLiteTypeMeta>();

    public StorIOSQLiteTypeMeta(String simpleName, String packageName, String tableName) {
        this.simpleName = simpleName;
        this.packageName = packageName;
        this.tableName = tableName;
    }
}

File: storio-sqlite-annotations-processor/src/main/kotlin/com/pushtorefresh/storio2/sqlite/annotations/processor/introspection/StorIOSQLiteTypeMeta.kt
package com.pushtorefresh.storio2.sqlite.annotations.processor.introspection

import com.pushtorefresh.storio2.common.annotations.processor.introspection.StorIOTypeMeta
import com.pushtorefresh.storio2.sqlite.annotations.StorIOSQLiteType

class StorIOSQLiteTypeMeta(
        simpleName: String,
        packageName: String,
        storIOType: StorIOSQLiteType,
        needsCreator: Boolean)
    : StorIOTypeMeta<StorIOSQLiteType, StorIOSQLiteColumnMeta>(
        simpleName,
        packageName,
        storIOType,
        needsCreator)